File: restore_cdn.py
import glob
import re

files = glob.glob('*.html')

for fname in files:
    with open(fname, 'r', encoding='utf-8') as f:
        c = f.read()

    # Replace local stylesheet with CDN
    c = c.replace('<link rel="stylesheet" href="style.css">', '<script src="https://cdn.tailwindcss.com"></script>')

    # Add tailwind.config script block if missing
    config_script = """
    <script>
        tailwind.config = {
            theme: {
                extend: {
                    fontFamily: {
                        sans: ['Poppins', 'sans-serif'],
                        serif: ['Cinzel', 'serif'],
                    },
                    colors: {
                        gold: {
                            50: '#fdfbf7',
                            100: '#fcf7e8',
                            400: '#e5c060',
                            500: '#D4AF37', /* Primary Gold */
                            600: '#bca02d',
                            700: '#967d22',
                        }
                    }
                }
            }
        }
    </script>"""

    if "tailwind.config =" not in c:
        # Insert before </head>
        c = c.replace('</head>', config_script + '\n</head>')

    with open(fname, 'w', encoding='utf-8') as f:
        f.write(c)
    
    print(f"Restored CDN in {fname}")

File: scripts/restore-cdn.js
/**
 * restore-cdn.js — Replaces restore_cdn.py
 * Swaps local style.css back to the Tailwind CDN script tag,
 * and re-injects the tailwind.config block before </head> if missing.
 *
 * Usage: node scripts/restore-cdn.js
 */

const fs     = require('fs');
const path   = require('path');
const glob   = require('glob');

// ── Load Tailwind config from tailwind.config.js ────────────────
// We read the theme directly from the existing config file so there
// is a single source of truth — no duplication.
const twConfig = require('../tailwind.config.js');
const { colors = {}, fontFamily = {} } = twConfig.theme?.extend ?? {};

// Build an inline tailwind.config script block from the JS config
const configScript = `
  <script>
    tailwind.config = {
      theme: {
        extend: {
          fontFamily: ${JSON.stringify(fontFamily, null, 10)},
          colors: ${JSON.stringify(colors, null, 10)}
        }
      }
    }
  </script>`;

// ── Process every root-level HTML file ───────────────────────────
const rootDir   = path.join(__dirname, '..');
const htmlFiles = glob.sync('*.html', { cwd: rootDir });

htmlFiles.forEach(fname => {
  const filePath = path.join(rootDir, fname);
  let content    = fs.readFileSync(filePath, 'utf-8');

  // Replace local stylesheet with CDN
  content = content.replace(
    /<link\s+rel="stylesheet"\s+href="style\.css"\s*>/g,
    '<script src="https://cdn.tailwindcss.com"></script>'
  );

  // Inject tailwind.config block before </head> if it's not already there
  if (!content.includes('tailwind.config =')) {
    content = content.replace('</head>', configScript + '\n</head>');
  }

  fs.writeFileSync(filePath, content, 'utf-8');
  console.log(`✅  Restored CDN in ${fname}`);
});

console.log('\n🎉  CDN restored in all files!');
